tests/integration/query/commits/with_delete_test.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package commits
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestQueryCommits_WithDelete(t *testing.T) {
```

Comment:
nitpick: the test name is not quite descriptive. Had to read the test's body in order to understand what it tests.
Wouldn't something like `TestQueryCommits_AfterDocDeletion_ShouldStillFetch` fit better?